src/cli/test/reporters/github/index.ts

```diff
@@ -0,0 +1,61 @@
+import type { Context, Reporter } from '@jest/reporters';
+import type { AggregatedResult, TestResult } from '@jest/test-result';
+
+import * as GitHub from '../../../../api/github';
+import {
+  buildNameFromEnvironment,
+  enabledFromEnvironment,
+} from '../../../../api/github/environment';
+
+import { createAnnotations } from './annotations';
+
+const DEFAULT_NAME = 'noDisplayName';
+
+export default class GitHubReporter implements Pick<Reporter, 'onRunComplete'> {
+  async onRunComplete(
```

Comment:
Does Jest fail the test run if a reporter throws? If so, we'll need to roll something like #649 ourselves.

Reply:
Good question - I should test that